monkey/infection_monkey/island_api_client/http_island_api_client.py

```diff
@@ -123,8 +129,6 @@ def get_agent_binary(self, operating_system: OperatingSystem) -> bytes:
     @handle_authentication_token_expiration
     def get_otp(self) -> str:
         response = self._http_client.get("/agent-otp")
-        if response.status_code == HTTPStatus.TOO_MANY_REQUESTS:
-            raise IslandAPIRequestLimitExceededError("Too many requests to get OTP.")
```

Comment:
Doesn't the `IslandAPIAgentOTPProvider` still expect this error to be raised?

Reply:
Oh, I see, it's now handled by the HTTPClient decorator

Reply:
It gets raised from `self._http_client.get()` in the `handle_island_errors()` decorator.